database/voter.go

```diff
@@ -0,0 +1,29 @@
+package database
+
+import (
+	"context"
+	"time"
+)
+
+type Voter struct {
+	Id     string   `bson:"_id"`
+	PollId string	`bson:"pollId"`
+	UserId string   `bson:"userId"`
+}
+
+func RecordVoter(voter *Voter) error {
+	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Second)
```

Comment:
Since you're introducing contexts, it'd not be a bad idea to include `ctx` as the first argument. This pushes the `context.TODO()` call off to the caller, but also makes it more visibile and makes it more likely that future editors will pass along contexts

Reply:
Max,

The `RecordVoter()` function is a carbon-copy of the `CastSimpleVote()` and `CastRankedVote()` functions from [simple_vote.go](https://github.com/ComputerScienceHouse/vote/blob/mistress/database/simple_vote.go) and [ranked_vote.go](https://github.com/ComputerScienceHouse/vote/blob/mistress/database/ranked_vote.go), barring a one-character change to the target database table ("votes" -> "voters").

I would be more than happy to refactor the signature of this function to take a context. However, since the very same code is present within the `CastSimpleVote()` and `CastRankedVote()` functions, it would most certainly be best to refactor those, too, for consistency sake.

Just confirming there are no objections to this plan before I pursue it.

Reply:
Go for it